GitUI/CommandsDialogs/SettingsDialog/MergeToolsHelper.cs

```diff
@@ -41,44 +41,49 @@ public static string FindFileInFolders(string fileName, params string[] location
 
                 string programFilesPath = Environment.GetEnvironmentVariable("ProgramFiles");
 
-                string path;
-
-                if (!string.IsNullOrEmpty(programFilesPath))
+                if (CheckFileExists(programFilesPath, location, fileName, out string fullName))
                 {
-                    path = Path.Combine(programFilesPath, location);
-                    if (Directory.Exists(path))
-                    {
-                        string fullName = Path.Combine(path, fileName);
-                        if (File.Exists(fullName))
-                        {
-                            return fullName;
-                        }
-                    }
+                    return fullName;
                 }
 
                 if (IntPtr.Size == 8
                     || (!string.IsNullOrEmpty(Environment.GetEnvironmentVariable("PROCESSOR_ARCHITEW6432"))))
                 {
                     programFilesPath = Environment.GetEnvironmentVariable("ProgramFiles(x86)");
 
-                    if (!string.IsNullOrEmpty(programFilesPath))
+                    if (CheckFileExists(programFilesPath, location, fileName, out fullName))
                     {
-                        path = Path.Combine(programFilesPath, location);
-                        if (Directory.Exists(path))
-                        {
-                            string fullName = Path.Combine(path, fileName);
-                            if (File.Exists(fullName))
-                            {
-                                return fullName;
-                            }
-                        }
+                        return fullName;
                     }
                 }
+
+                string localAppDataPath = Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData);
+
+                if (CheckFileExists(localAppDataPath, location, fileName, out fullName))
+                {
+                    return fullName;
+                }
             }
 
             return string.Empty;
         }
 
+        private static bool CheckFileExists(string path, string location, string fileName, out string fullName)
+        {
+            if (!string.IsNullOrEmpty(path))
```

Comment:
please invert to reduce nesting, this make the code more readable

Reply:
Also join up nested `if`s in lines 49-54

Reply:
Done.